src/adjust/adjustfiltering.cpp

```diff
@@ -3,158 +3,91 @@
-  // sorted. Let's check this in debug builds.
+void AdjustFiltering::denyField(const QString& field, const QString& param) {
+  denyList.insert(field, param);
 
-  for (int i = 1; i < allowList.length(); ++i)
-    Q_ASSERT(allowList[i] > allowList[i - 1]);
+  if (allowList.contains(field)) {
+    allowList.remove(field);
+  }
+
+  if (mirrorList.contains(field)) {
+    mirrorList.remove(field);
+  }
+}
+
+void AdjustFiltering::mirrorField(const QString& field,
+                                  const MirrorParam& param) {
+  mirrorList.insert(field, param);
 
-  for (int i = 1; i < denyList.length(); ++i)
-    Q_ASSERT(denyList[i].m_paramName > denyList[i - 1].m_paramName);
+  if (allowList.contains(field)) {
+    allowList.remove(field);
+  }
 
-  for (int i = 1; i < mirrorList.length(); ++i) {
-    Q_ASSERT(mirrorList[i].m_paramName > mirrorList[i - 1].m_paramName);
-    // Only allowed params can be mirrored.
-    Q_ASSERT(allowList.contains(mirrorList[i].m_paramName));
+  if (denyList.contains(field)) {
+    denyList.remove(field);
   }
-#endif
+}
+
+QUrlQuery AdjustFiltering::filterParameters(QUrlQuery& parameters,
+                                            QStringList& unknownParameters) {
+  QUrlQuery newParameters;
 
   for (const QPair<QString, QString>& parameter : parameters.queryItems()) {
-    if (std::binary_search(allowList.begin(), allowList.end(),
-                           parameter.first)) {
+    if (allowList.contains(parameter.first)) {
       newParameters.addQueryItem(parameter.first, parameter.second);
       continue;
     }
 
-    const DenyParam* denyParam = std::lower_bound(
-        denyList.begin(), denyList.end(), parameter.first,
-        [](const DenyParam& data, const QString& value) -> bool {
-          return data.m_paramName < value;
-        });
-    if (denyParam && denyParam != denyList.end() &&
-        denyParam->m_paramName == parameter.first) {
-      newParameters.addQueryItem(parameter.first, denyParam->m_defaultValue);
+    if (denyList.contains(parameter.first)) {
+      newParameters.addQueryItem(parameter.first,
+                                 denyList.value(parameter.first));
       continue;
     }
 
-    const MirrorParam* mirrorParam = std::lower_bound(
-        mirrorList.begin(), mirrorList.end(), parameter.first,
-        [](const MirrorParam& data, const QString& value) -> bool {
-          return data.m_paramName < value;
-        });
-    if (mirrorParam && mirrorParam != mirrorList.end() &&
-        mirrorParam->m_paramName == parameter.first) {
+    if (mirrorList.contains(parameter.first)) {
+      const MirrorParam mirrorParam = mirrorList.value(parameter.first);
       bool found = false;
       for (const QPair<QString, QString>& otherParameter :
            parameters.queryItems()) {
-        if (mirrorParam->m_mirrorParamName == otherParameter.first) {
+        if (mirrorParam.m_mirrorParamName == otherParameter.first) {
```

Comment:
Move the `if` outside the `for`. In this way, you run the `for`-loop only if the `mirrorParam.m_mirrorParamName` is allowed.